src/main/java/com/box/sdk/BoxSearch.java

```diff
@@ -0,0 +1,63 @@
+package com.box.sdk;
+import java.net.URL;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+/**
+ * Represents search on Box. This class can be used to search through your box instance.
+ * In addition this lets you take advantage of all the advanced search features.
+ *
+ * <p>Unless otherwise noted, the methods in this class can throw an unchecked {@link BoxAPIException} (unchecked
+ * meaning that the compiler won't force you to handle it) if an error occurs. If you wish to implement custom error
+ * handling for errors related to the Box REST API, you should capture this exception explicitly.</p>
+ */
+public class BoxSearch {
+
+    private static final URLTemplate SEARCH_URL_TEMPLATE = new URLTemplate("search");
+    private final BoxAPIConnection api;
+
+    /**
+     * Constructs a Search to be used by everything.
+     * @param  api the API connection to be used by the search.
+     */
+    public BoxSearch(BoxAPIConnection api) {
+        this.api = api;
+    }
+
+    /**
+     * Searches all descendant folders using a given query and query parameters.
+     * @param  offset is the starting position.
+     * @param  limit the number of search results CANNONT Exceed 1000.
+     * @param  bsp containing query and advanced search capabilities.
+     * @return a PartialCollection containing the search results.
+     */
+    public PartialCollection<BoxItem.Info> search(long offset, long limit, final BoxSearchParameters bsp) {
```

Comment:
Rename to `searchRange` to be more consistent with `BoxFolder`. That way we can add another search method later which can do automatic paging.